packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -16,6 +19,47 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// Returns list of all available cameras and their descriptions.
   @override
   Future<List<CameraDescription>> availableCameras() async {
-    throw UnimplementedError('availableCameras() is not implemented.');
+    final List<CameraDescription> cameraDescriptions = <CameraDescription>[];
+
+    ProcessCameraProvider processCameraProvider =
+        await ProcessCameraProvider.getInstance();
+    final List<CameraInfo> cameraInfos =
+        await processCameraProvider!.getAvailableCameraInfos();
+
+    final CameraSelector backCameraSelector =
+        CameraSelector.getDefaultBackCamera();
+    final CameraSelector frontCameraSelector =
+        CameraSelector.getDefaultFrontCamera();
+
+    CameraLensDirection cameraLensDirection;
+    int cameraCount = 0;
+    int cameraSensorOrientation;
+    String cameraName;
+
+    for (final CameraInfo cameraInfo in cameraInfos) {
+      // Determine the lens direction by filtering the CameraInfo
+      // TODO(gmackall): replace this with call to CameraInfo.getLensFacing when changes containing that method are available
+      if ((await backCameraSelector.filter(<CameraInfo>[cameraInfo]))
+          .isNotEmpty) {
+        cameraLensDirection = CameraLensDirection.back;
+      } else if ((await frontCameraSelector.filter(<CameraInfo>[cameraInfo]))
+          .isNotEmpty) {
+        cameraLensDirection = CameraLensDirection.front;
+      } else {
+        //Skip this CameraInfo as its lens direction is unknown
+        continue;
```

Comment:
What cameras would this case catch? I assume external cameras, but I was under the impression that those are not supported by CameraX right now. For what it's worth, this makes sense, but just wondering if this is case we'd actually hit.

Reply:
I believe it won't catch any case at the moment, but in addition to the upcoming changes to add support for external cameras, they are adding a `LENS_FACING_UNKNOWN` (see [here](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:camera/camera-core/src/main/java/androidx/camera/core/CameraSelector.java;l=45?q=file:androidx%2Fcamera%2Fcore%2FCameraSelector.java%20class:androidx.camera.core.CameraSelector&ss=androidx%2Fplatform%2Fframeworks%2Fsupport)). 

We will probably want to change this code anyway because the change to expose lens facing info in CameraInfo will happen around the same time, but I wanted to add this just so nothing slips through.